Error thrown by setting threshold not being picked up by lint-staged

**Describe the bug**
I'm trying to hook up jscpd with lint-staged and husky to fail a commit. When I added some duplicated code, jscpd throws a `UnhandledPromiseRejectionWarning:` error, but does not prevent the lint-staged execution from failing as expected. I'm not sure if this is a bug with jscpd or me just not understanding how to configure these three packages to work together.


My package.json has this in it's scripts section
```
"scripts": {
   "check-for-dupes": "jscpd ./src -t 1",
   "lint": "yarn lintSyntax && yarn check-for-dupes"
},
  "husky": {
    "hooks": {
      "pre-commit": "lint-staged",
      "pre-push": "yarn check-for-dupes"
    }
  },
  "lint-staged": {
    "src/**/*.{js,jsx,css,scss}": [
      "yarn lint"
    ]
  },
```

**To Reproduce**
Running `yarn check-for-dupes` results in the expected threshold violation being triggered
```
~/S/sampleRepo (master↑8|●1✚1) $ yarn check-for-dupes                                                                                                                                                                                                                                         16:27:06
yarn run v1.16.0
$ jscpd ./src -t 1 -r console
Clone found (javascript):
 - src/Settings.js [57:13 - 70:9]
   src/Settings.js [28:12 - 41:26]

┌────────────┬────────────────┬─────────────┬──────────────┬──────────────────┬───────┐
│ Format     │ Files analyzed │ Total lines │ Clones found │ Duplicated lines │ %     │
├────────────┼────────────────┼─────────────┼──────────────┼──────────────────┼───────┤
│ javascript │ 9              │ 664         │ 1            │ 13               │ 1.96% │
├────────────┼────────────────┼─────────────┼──────────────┼──────────────────┼───────┤
│ Total:     │ 9              │ 664         │ 1            │ 13               │ 1.96% │
└────────────┴────────────────┴─────────────┴──────────────┴──────────────────┴───────┘
Found 1 clones.
ERROR: jscpd found too many duplicates (1.96%) over threshold (1%)
(node:58977) UnhandledPromiseRejectionWarning: Error: ERROR: jscpd found too many duplicates (1.96%) over threshold (1%)
    at ThresholdReporter.report (/Users/myUser/Source/sampleRepo/node_modules/@jscpd/finder/dist/reporters/threshold.js:12:19)
    at /Users/myUser/Source/sampleRepo/node_modules/@jscpd/finder/dist/in-files-detector.js:76:26
    at Array.forEach (<anonymous>)
    at /Users/myUser/Source/sampleRepo/node_modules/@jscpd/finder/dist/in-files-detector.js:75:28
(node:58977) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2754)
(node:58977) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
✨  Done in 0.67s.
```

Running `yarn lint-staged` does not error out as expected
```
$ /Users/myUser/Source/sampleRepo/node_modules/.bin/lint-staged
✔ Preparing...
✔ Running tasks...
✔ Applying modifications...
✔ Cleaning up... 
✨  Done in 2.96s.
~/S/sampleRepo (master↑8|●1✚1) $
```

**Expected behavior**
I also have eslint hooked up to lint-staged, and this behaves as expected when a linting error is introduced.
Running `yarn lint-staged `

```
yarn run v1.16.0
$ /Users/myUser/Source/sampleRepo/node_modules/.bin/lint-staged
✔ Preparing...
⚠ Running tasks...
  ❯ Running tasks for src/**/*.{js,jsx,css,scss}
    ✖ yarn lint [FAILED]
↓ Skipped because of errors from tasks. [SKIPPED]
✔ Reverting to original state because of errors...
✔ Cleaning up... 

✖ yarn lint:
error Command failed with exit code 1.
error Command failed with exit code 1.
$ yarn lintSyntax && yarn check-for-dupes /Users/myUser/Source/sampleRepo/src/Settings.js
$ eslint '**/*.js' --fix --ignore-path .gitignore --ignore-pattern '!.eslintrc.js'

/Users/myUser/Source/sampleRepo/src/Settings.js
  44:10  error  Unexpected constant condition  no-constant-condition

✖ 1 problem (1 error, 0 warnings)

info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
```

**Screenshots**
Included code snippets above

**Desktop (please complete the following information):**
 - OS: macOs Mojave
 - OS Version v10.14.6
 - NodeJS Version v12.5.0
 - jscpd version ^3.3.0-rc.6


**Additional context**
Add any other context about the problem here.

--- a/packages/jscpd/bin/jscpd
+++ b/packages/jscpd/bin/jscpd
@@ -12,7 +12,10 @@ if (wantsCompiled || !devMode) {
     // this runs from the compiled javascript source
     var jscpd = require(`${__dirname}/../dist`);
     try {
-    	jscpd.jscpd(process.argv);
+    	jscpd.jscpd(process.argv).catch((e) => {
+        console.log(e);
+        process.exit(1);
+      });
     } catch (e) {
     	console.log(e);
     	process.exit(1);
@@ -24,7 +27,10 @@ if (wantsCompiled || !devMode) {
     // run the CLI with the current process arguments
     var jscpd = require(`${__dirname}/../src`);
     try {
-    	jscpd.jscpd(process.argv);
+    	jscpd.jscpd(process.argv).catch((e) => {
+        console.log(e);
+        process.exit(1);
+      });
     } catch (e) {
     	console.log(e);
     	process.exit(1);

